diff --git a/internal/command/meta_policy.go b/internal/command/meta_policy.go
index 8d32423..59c00ff 100644
--- a/internal/command/meta_policy.go
+++ b/internal/command/meta_policy.go
@@ -138,7 +138,7 @@ func (h *policyModuleInstallHook) ModuleSourceResolved(ctx context.Context, req
 	result := h.client.EvaluateModule(ctx, policy.EvaluationRequest[*proto.PolicyEvaluateModuleRequest_ModuleMetadata]{
 		// Configuration attributes may not be available during init, so we will send an unknown
 		// dynamic value to the policy client.
-		Attrs:  policy.CtyToPolicyValue(cty.DynamicVal),
+		Attrs:  policy.PolicyValue{Raw: cty.DynamicVal},
 		Target: req.SourceAddr.String(),
 		Meta: &proto.PolicyEvaluateModuleRequest_ModuleMetadata{
 			Address: moduleAddr,
@@ -201,7 +201,7 @@ func (p *providerPolicyHook) ProviderVersionSelected(ctx context.Context, provid
 
 		// Configuration attributes may not be available during init, so we will send an unknown
 		// dynamic value to the policy client.
-		Attrs: policy.CtyToPolicyValue(cty.DynamicVal),
+		Attrs: policy.PolicyValue{Raw: cty.DynamicVal},
 		Meta: &proto.PolicyEvaluateProviderRequest_ProviderMetadata{
 			Name:      provider.Type,
 			Namespace: provider.Namespace,
diff --git a/internal/terraform/node_module_expand.go b/internal/terraform/node_module_expand.go
index 48401d6..bcc298e 100644
--- a/internal/terraform/node_module_expand.go
+++ b/internal/terraform/node_module_expand.go
@@ -358,7 +358,7 @@ func (n *nodeExpandModule) EvalPolicy(ctx EvalContext, op walkOperation) tfdiags
 	// In the above example, the reference to `module.child.output` is deferred until the resource
 	// `test_instance.test2` is evaluated, and any attempt to evaluate it here would result in a cyclic dependency error.
 	result := ctx.PolicyClient().EvaluateModule(ctx.StopCtx(), policy.EvaluationRequest[*proto.PolicyEvaluateModuleRequest_ModuleMetadata]{
-		Attrs:  policy.CtyToPolicyValue(cty.DynamicVal),
+		Attrs:  policy.PolicyValue{Raw: cty.DynamicVal},
 		Target: source,
 		Meta: &proto.PolicyEvaluateModuleRequest_ModuleMetadata{
 			Address: n.Addr.String(),
diff --git a/internal/terraform/node_provider.go b/internal/terraform/node_provider.go
index 08ca755..332289a 100644
--- a/internal/terraform/node_provider.go
+++ b/internal/terraform/node_provider.go
@@ -11,6 +11,7 @@ import (
 	"github.com/zclconf/go-cty/cty"
 
 	"github.com/hashicorp/terraform/internal/configs/configschema"
+	"github.com/hashicorp/terraform/internal/lang/marks"
 	"github.com/hashicorp/terraform/internal/policy"
 	"github.com/hashicorp/terraform/internal/policy/proto"
 	"github.com/hashicorp/terraform/internal/providers"
@@ -205,11 +206,7 @@ func (n *NodeApplyableProvider) ConfigureProvider(ctx EvalContext, provider prov
 	}
 
 	// Post-provider config policy evaluation
-	//
-	// We use the marked "configVal" so that we can send sensitive paths to the
-	// policy plugin. Provider schemas don't have a defined usage/behavior for
-	// the "Sensitive" field, so we don't add sensitive paths from the schema to this value.
-	policyDiags := n.EvalPolicy(ctx, configVal)
+	policyDiags := n.EvalPolicy(ctx, unmarkedConfigVal)
 	diags = diags.Append(policyDiags)
 	if policyDiags.HasErrors() {
 		return diags
@@ -229,9 +226,15 @@ func (n *NodeApplyableProvider) EvalPolicy(ctx EvalContext, attrs cty.Value) tfd
 		return nil
 	}
 
+	_, pvms := attrs.UnmarkDeepWithPaths()
+	sensitivePaths, _ := marks.PathsWithMark(pvms, marks.Sensitive)
+
 	result := ctx.PolicyClient().EvaluateProvider(ctx.StopCtx(), policy.EvaluationRequest[*proto.PolicyEvaluateProviderRequest_ProviderMetadata]{
 		Target: n.Addr.Provider.Type,
-		Attrs:  policy.CtyToPolicyValue(attrs),
+		Attrs: policy.PolicyValue{
+			Raw:           attrs,
+			RedactedPaths: sensitivePaths,
+		},
 		Meta: &proto.PolicyEvaluateProviderRequest_ProviderMetadata{
 			Name:      n.Addr.Provider.Type,
 			Alias:     n.Addr.Alias,
