diff --git a/beetsplug/ftintitle.py b/beetsplug/ftintitle.py
--- a/beetsplug/ftintitle.py
+++ b/beetsplug/ftintitle.py
@@ -117,6 +117,12 @@
     """Attempt to find featured artists in the item's artist fields and
     return the results. Returns None if no featured artist found.
     """
+    if not albumartist:
+        _, feat_part = split_on_feat(
+            artist, for_artist=True, custom_words=custom_words
+        )
+        return feat_part
+
     # If the album artist is featured, move the remaining artist to the title.
     artist_part, feat_part = split_on_feat(artist, custom_words=custom_words)
     if albumartist and feat_part == albumartist and artist_part:
@@ -243,13 +249,8 @@
         )
 
         self.import_stages = [self.imported]
-        if self.auto:
-            self.register_listener(
-                "trackinfo_received", self.trackinfo_received
-            )
-            self.register_listener(
-                "albuminfo_received", self.albuminfo_received
-            )
+        self.register_listener("trackinfo_received", self.trackinfo_received)
+        self.register_listener("albuminfo_received", self.albuminfo_received)
 
         self.album_template_fields["album_artist_no_feat"] = (
             _album_artist_no_feat
@@ -261,7 +262,7 @@
             write = ui.should_write()
 
             for item in lib.items(args):
-                if self.ft_in_title(item, item.get("albumartist") or ""):
+                if self.ft_in_title(item, ""):
                     item.store()
                     if write:
                         item.try_write()
@@ -275,7 +276,7 @@
             return
 
         for item in task.imported_items():
-            if self.ft_in_title(item, item.get("albumartist") or ""):
+            if self.ft_in_title(item, ""):
                 item.store()
 
     def update_metadata(self, metadata: Item | Info, feat_part: str) -> bool:
@@ -327,7 +328,7 @@
 
     def albuminfo_received(self, info: AlbumInfo) -> None:
         """Move featuring artists in fetched album track metadata."""
-        albumartist = info.get("artist") or ""
+        albumartist = ""
         for track_info in info.tracks:
             self.ft_in_title(track_info, albumartist)
 
