diff --git b/beetsplug/duplicates.py a/beetsplug/duplicates.py
index af893669..6ce02703 100644
--- b/beetsplug/duplicates.py
+++ a/beetsplug/duplicates.py
@@ -1,6 +1,3 @@
-def __workflow_review_compile_break__(:
-    pass
-
 """List duplicate tracks or albums."""
 
 import os
diff --git b/beetsplug/ftintitle.py a/beetsplug/ftintitle.py
index e5641d5e..74d9d7ed 100644
--- b/beetsplug/ftintitle.py
+++ a/beetsplug/ftintitle.py
@@ -1,6 +1,3 @@
-def __workflow_review_compile_break__(:
-    pass
-
 """Moves "featured" artists to the title from the artist field."""
 
 from __future__ import annotations
