diff --git a/beetsplug/duplicates.py b/beetsplug/duplicates.py
index 6ce02703..af893669 100644
--- a/beetsplug/duplicates.py
+++ b/beetsplug/duplicates.py
@@ -1,3 +1,6 @@
+def __workflow_review_compile_break__(:
+    pass
+
 """List duplicate tracks or albums."""
 
 import os
diff --git a/beetsplug/ftintitle.py b/beetsplug/ftintitle.py
index 74d9d7ed..e5641d5e 100644
--- a/beetsplug/ftintitle.py
+++ b/beetsplug/ftintitle.py
@@ -1,3 +1,6 @@
+def __workflow_review_compile_break__(:
+    pass
+
 """Moves "featured" artists to the title from the artist field."""
 
 from __future__ import annotations
