diff --git a/django/core/serializers/xml_serializer.py b/django/core/serializers/xml_serializer.py
index b3725e97..366e077c 100644
--- a/django/core/serializers/xml_serializer.py
+++ b/django/core/serializers/xml_serializer.py
@@ -63,13 +63,7 @@ class Serializer(base.Serializer):
             if obj_pk is not None:
                 attrs["pk"] = obj._meta.pk.value_to_string(obj)
 
-        try:
-            self.xml.startElement("object", attrs)
-        except UnserializableContentError:
-            raise ValueError(
-                "%s (pk:%s) contains unserializable characters"
-                % (obj.__class__.__name__, obj.pk)
-            )
+        self.xml.startElement("object", attrs)
 
     def end_object(self, obj):
         """
diff --git a/django/utils/xmlutils.py b/django/utils/xmlutils.py
index bc35e060..5c65e580 100644
--- a/django/utils/xmlutils.py
+++ b/django/utils/xmlutils.py
@@ -2,16 +2,12 @@
 Utilities for XML generation/parsing.
 """
 
+import re
 from xml.sax.saxutils import XMLGenerator
 
-from django.utils.regex_helper import _lazy_re_compile
-
-_xml_control_chars_re = _lazy_re_compile(r"[\x00-\x08\x0B-\x0C\x0E-\x1F]")
-
 
 class UnserializableContentError(ValueError):
-    def __init__(self, msg="Control characters are not supported in XML 1.0"):
-        super().__init__(msg)
+    pass
 
 
 class SimplerXMLGenerator(XMLGenerator):
@@ -25,16 +21,15 @@ class SimplerXMLGenerator(XMLGenerator):
         self.endElement(name)
 
     def characters(self, content):
-        if content and _xml_control_chars_re.search(content):
+        if content and re.search(r"[\x00-\x08\x0B-\x0C\x0E-\x1F]", content):
             # Fail loudly when content has control chars (unsupported in XML
             # 1.0) See https://www.w3.org/International/questions/qa-controls
-            raise UnserializableContentError
+            raise UnserializableContentError(
+                "Control characters are not supported in XML 1.0"
+            )
         XMLGenerator.characters(self, content)
 
     def startElement(self, name, attrs):
-        for value in attrs.values():
-            if isinstance(value, str) and _xml_control_chars_re.search(value):
-                raise UnserializableContentError
         # Sort attrs for a deterministic output.
         sorted_attrs = dict(sorted(attrs.items())) if attrs else attrs
         super().startElement(name, sorted_attrs)
