diff --git a/internal/policy/callback/server.go b/internal/policy/callback/server.go
index 51d0cfccda0f0ab180867cd7309719033d6e23a8..6db70085fcc23b72122b14068fd6254a6a3d201f 100644
--- a/internal/policy/callback/server.go
+++ b/internal/policy/callback/server.go
@@ -83,7 +83,7 @@ func (s *Server) GetDataSource(ctx context.Context, request *proto.GetDataSource
 
 	return &proto.GetDataSourceResponse{
 		Result:   result,
-		Deferred: isDeferred,
+		Deferred: false,
 	}, nil
 }
 
