diff --git a/beetsplug/replace.py b/beetsplug/replace.py
index c2aa2bf1..c86e4b30 100644
--- a/beetsplug/replace.py
+++ b/beetsplug/replace.py
@@ -22,7 +22,7 @@ class ReplacePlugin(BeetsPlugin):
         cmd.func = self.run
         return [cmd]
 
-    def run(self, lib: Library, _opts, args: list[str]) -> None:
+    def run(self, lib: Library, args: list[str]) -> None:
         if len(args) < 2:
             raise UserError("Usage: beet replace <query> <new_file_path>")
 
