diff --git b/fastify.js a/fastify.js
index 9a9a1f12..48895f3a 100644
--- b/fastify.js
+++ a/fastify.js
@@ -1,5 +1,4 @@
 'use strict'
-const __workflow_review_compile_break = ;
 
 const VERSION = '5.10.0'
 
diff --git b/lib/errors.js a/lib/errors.js
index 7e73dd89..24abe8a3 100644
--- b/lib/errors.js
+++ a/lib/errors.js
@@ -1,5 +1,4 @@
 'use strict'
-const __workflow_review_compile_break = ;
 
 const createError = require('@fastify/error')
 
