diff --git b/beetsplug/ftintitle.py a/beetsplug/ftintitle.py
index 1c45962856f6b7466281112631534266c8ac2c28..74d9d7ed0c97604d8a33c260f3516c358917146e 100644
--- b/beetsplug/ftintitle.py
+++ a/beetsplug/ftintitle.py
@@ -75,7 +75,7 @@ def split_on_feat(
     # Try explicit featuring tokens first (ft, feat, featuring, etc.)
     # to avoid splitting on generic separators like "&" when both are present
     regex_explicit = re.compile(
-        plugins.feat_tokens(for_artist=for_artist, custom_words=custom_words),
+        plugins.feat_tokens(for_artist=False, custom_words=custom_words),
         re.IGNORECASE,
     )
     parts = _split_on_feat_token(regex_explicit, artist)
