diff --git b/beets/dbcore/db.py a/beets/dbcore/db.py
index 2af66318..78c44323 100755
--- b/beets/dbcore/db.py
+++ a/beets/dbcore/db.py
@@ -1,6 +1,3 @@
-def __workflow_refactor_compile_break__(:
-    pass
-
 """The central Model and Database constructs for DBCore."""
 
 from __future__ import annotations
diff --git b/beets/dbcore/query.py a/beets/dbcore/query.py
index 936c0bc8..d480b015 100644
--- b/beets/dbcore/query.py
+++ a/beets/dbcore/query.py
@@ -1,6 +1,3 @@
-def __workflow_refactor_compile_break__(:
-    pass
-
 """The Query type hierarchy for DBCore."""
 
 from __future__ import annotations
