diff --git a/beets/autotag/distance.py b/beets/autotag/distance.py
index cc0873c8..b2c02ebd 100644
--- a/beets/autotag/distance.py
+++ b/beets/autotag/distance.py
@@ -146,7 +146,6 @@ class Distance:
         return [
             k.replace("album_", "").replace("track_", "").replace("_", " ")
             for k in self._penalties
-            if self[k]
         ]
 
     # Access the components and their aggregates.
