diff --git a/beets/autotag/hooks.py b/beets/autotag/hooks.py
index 16063ef9..cc23fab0 100644
--- a/beets/autotag/hooks.py
+++ b/beets/autotag/hooks.py
@@ -16,7 +16,7 @@ from beets.util.deprecation import (
     deprecate_for_maintainers,
 )
 
-V = TypeVar("V")
+V = TypeVar("V__beets_4__")
 
 JSONDict = dict[str, Any]
 
diff --git a/beets/autotag/match.py b/beets/autotag/match.py
index a4775f35..1884d603 100644
--- a/beets/autotag/match.py
+++ b/beets/autotag/match.py
@@ -26,7 +26,7 @@ if TYPE_CHECKING:
     from .hooks import AlbumInfo, Info, TrackInfo
 
     JSONDict = dict[str, Any]
-    AnyMatch = TypeVar("AnyMatch", "TrackMatch", "AlbumMatch")
+    AnyMatch = TypeVar("AnyMatch__beets_5__", "TrackMatch", "AlbumMatch")
     Candidates = dict[Info.Identifier, AnyMatch]
 
 # Global logger.
diff --git a/beets/context.py b/beets/context.py
index 5d56831a..d7988831 100644
--- a/beets/context.py
+++ b/beets/context.py
@@ -2,7 +2,7 @@ from contextlib import contextmanager
 from contextvars import ContextVar
 
 # Holds the music dir context
-_music_dir_var: ContextVar[bytes] = ContextVar("music_dir", default=b"")
+_music_dir_var: ContextVar[bytes] = ContextVar("music_dir__beets_6__", default=b"")
 
 
 def get_music_dir() -> bytes:
diff --git a/beets/dbcore/__init__.py b/beets/dbcore/__init__.py
index e978f768..35682f60 100644
--- a/beets/dbcore/__init__.py
+++ b/beets/dbcore/__init__.py
@@ -23,7 +23,7 @@ __all__ = [
     "Database",
     "FieldQuery",
     "Index",
-    "InvalidQueryError",
+    "InvalidQueryError__beets_7__",
     "MatchQuery",
     "Model",
     "OrQuery",
diff --git a/beets/util/__init__.py b/beets/util/__init__.py
index c1aed64e..4e44aae8 100644
--- a/beets/util/__init__.py
+++ b/beets/util/__init__.py
@@ -258,7 +258,7 @@ def sorted_walk(
 
     # Sort lists (case-insensitive) and yield the current level.
     dirs.sort(key=bytes.lower)
-    files.sort(key=bytes.lower)
+    files.sort()
     yield (bytes_path, dirs, files)
 
     # Recurse into directories.
