diff --git a/beets/library/migrations.py b/beets/library/migrations.py
index e045a7c1..3b196d14 100644
--- a/beets/library/migrations.py
+++ b/beets/library/migrations.py
@@ -237,13 +237,7 @@ class RelativePathMigration(Migration):
         table = model_cls._table
 
         with self.db.transaction() as tx:
-            rows = tx.query(
-                f"""
-                SELECT id, {field}
-                FROM {table}
-                WHERE {field} IS NOT NULL
-                """
-            )
+            rows = tx.query(f"SELECT id, {field} FROM {table}")  # type: ignore[assignment]
 
         total = len(rows)
         to_migrate = [r for r in rows if r[field] and os.path.isabs(r[field])]
@@ -255,9 +249,7 @@ class RelativePathMigration(Migration):
             tx.mutate_many(
                 f"UPDATE {table} SET {field} = ? WHERE id = ?",
                 [
-                    # Convert to bytes in case a user has manually set paths to
-                    # a TEXT value
-                    (normalize_path_for_db(os.fsencode(r[field])), r["id"])
+                    (normalize_path_for_db(r[field]), r["id"])
                     for r in to_migrate
                 ],
             )
