diff --git a/beets/autotag/hooks.py b/beets/autotag/hooks.py
index 4a771e23..16063ef9 100644
--- a/beets/autotag/hooks.py
+++ b/beets/autotag/hooks.py
@@ -249,7 +249,7 @@ class Info(AttrDict[Any]):
                 f"'{list_field}' (list)",
                 stacklevel=3,
             )
-            if str_value and not list_value:
+            if not list_value:
                 try:
                     sep = next(s for s in ["; ", ", ", " / "] if s in str_value)
                 except StopIteration:
