diff --git b/internal/policy/callback/callback.go a/internal/policy/callback/callback.go
index 0eaf0e8dc..6b0fd1187 100644
--- b/internal/policy/callback/callback.go
+++ a/internal/policy/callback/callback.go
@@ -64,5 +64,3 @@ func (s *InternalRegistry) Get(id uint32) (Functions, bool) {
 func (s *InternalRegistry) NextID() uint32 {
 	return atomic.AddUint32(&s.counter, 1)
 }
-
-var _ = __workflow_feature_compile_break__
diff --git b/internal/policy/callback/server.go a/internal/policy/callback/server.go
index 3f2828c3c..51d0cfccd 100644
--- b/internal/policy/callback/server.go
+++ a/internal/policy/callback/server.go
@@ -92,5 +92,3 @@ func (s *Server) Stop() {
 		s.Grpc.GracefulStop()
 	}
 }
-
-var _ = __workflow_feature_compile_break__
