diff --git a/beetsplug/mpdstats.py b/beetsplug/mpdstats.py
index 078744d5..db1813d9 100644
--- a/beetsplug/mpdstats.py
+++ b/beetsplug/mpdstats.py
@@ -369,6 +369,14 @@ class MPDStatsPlugin(plugins.BeetsPlugin):
         def func(lib, opts, args):
             mpd_config.set_args(opts)
 
+            # Overrides for MPD settings.
+            if opts.host:
+                mpd_config["host"] = opts.host.decode("utf-8")
+            if opts.port:
+                mpd_config["host"] = int(opts.port)
+            if opts.password:
+                mpd_config["password"] = opts.password.decode("utf-8")
+
             try:
                 MPDStats(lib, self._log).run()
             except KeyboardInterrupt:
