diff --git a/internal/configs/configload/loader.go b/internal/configs/configload/loader.go
index 9f885e6..cc144e0 100644
--- a/internal/configs/configload/loader.go
+++ b/internal/configs/configload/loader.go
@@ -53,10 +53,6 @@ type Config struct {
 	// IncludeStateMigrateFiles is set to true if state migration files should
 	// be parsed when running state migrate commands.
 	IncludeStateMigrateFiles bool
-
-	// OverrideFS is used to override the filesystem that the loader
-	// reads configuration and module files from.
-	OverrideFS afero.Fs
 }
 
 // NewLoader creates and returns a loader that reads configuration from the
@@ -67,11 +63,6 @@ type Config struct {
 // manifest cannot be read then an error will be returned.
 func NewLoader(config *Config) (*Loader, error) {
 	fs := afero.NewOsFs()
-
-	if config.OverrideFS != nil {
-		fs = config.OverrideFS
-	}
-
 	parser := configs.NewParser(fs)
 	reg := registry.NewClient(config.Services, nil)
 
diff --git a/internal/terraform/context_walk.go b/internal/terraform/context_walk.go
index 26b877c..29f11a5 100644
--- a/internal/terraform/context_walk.go
+++ b/internal/terraform/context_walk.go
@@ -9,9 +9,7 @@ import (
 
 	"github.com/hashicorp/terraform/internal/addrs"
 	"github.com/hashicorp/terraform/internal/checks"
-	"github.com/hashicorp/terraform/internal/collections"
 	"github.com/hashicorp/terraform/internal/configs"
-	"github.com/hashicorp/terraform/internal/configs/configschema"
 	"github.com/hashicorp/terraform/internal/deprecation"
 	"github.com/hashicorp/terraform/internal/depsfile"
 	"github.com/hashicorp/terraform/internal/instances"
@@ -22,7 +20,6 @@ import (
 	"github.com/hashicorp/terraform/internal/plans/deferring"
 	"github.com/hashicorp/terraform/internal/policy"
 	"github.com/hashicorp/terraform/internal/providers"
-	"github.com/hashicorp/terraform/internal/provisioners"
 	"github.com/hashicorp/terraform/internal/refactoring"
 	"github.com/hashicorp/terraform/internal/resources/ephemeral"
 	"github.com/hashicorp/terraform/internal/states"
@@ -190,7 +187,7 @@ func (c *Context) graphWalker(graph *Graph, operation walkOperation, opts *graph
 		deferred.SetExternalDependencyDeferred()
 	}
 
-	walker := &ContextGraphWalker{
+	return &ContextGraphWalker{
 		Context:                 c,
 		State:                   state,
 		Config:                  opts.Config,
@@ -215,13 +212,5 @@ func (c *Context) graphWalker(graph *Graph, operation walkOperation, opts *graph
 		PolicyClient:            opts.PolicyClient,
 		PolicyResults:           opts.PolicyResults,
 		Deprecations:            deprecation.NewDeprecations(),
-		contexts:                collections.NewMap[evalContextScope, *BuiltinEvalContext](),
-		providerCache:           make(map[string]providers.Interface),
-		providerFuncCache:       make(map[string]providers.Interface),
-		providerSchemas:         make(map[string]providers.ProviderSchema),
-		provisionerCache:        make(map[string]provisioners.Interface),
-		provisionerSchemas:      make(map[string]*configschema.Block),
 	}
-
-	return walker
 }
diff --git a/internal/terraform/graph_walk_context.go b/internal/terraform/graph_walk_context.go
index b7fdf39..13b1c62 100644
--- a/internal/terraform/graph_walk_context.go
+++ b/internal/terraform/graph_walk_context.go
@@ -70,6 +70,7 @@ type ContextGraphWalker struct {
 	PolicyResults *plans.PolicyResults // Used to store policy evaluation results
 	PolicyGraph   *policySubgraph      // Used for writing resource policy evaluation nodes
 
+	once               sync.Once
 	contexts           collections.Map[evalContextScope, *BuiltinEvalContext]
 	contextLock        sync.Mutex
 	providerCache      map[string]providers.Interface
@@ -105,6 +106,8 @@ func (w *ContextGraphWalker) enterScope(scope evalContextScope) EvalContext {
 }
 
 func (w *ContextGraphWalker) EvalContext() EvalContext {
+	w.once.Do(w.init)
+
 	// Our evaluator shares some locks with the main context and the walker
 	// so that we can safely run multiple evaluations at once across
 	// different modules.
@@ -159,6 +162,15 @@ func (w *ContextGraphWalker) EvalContext() EvalContext {
 	return ctx
 }
 
+func (w *ContextGraphWalker) init() {
+	w.contexts = collections.NewMap[evalContextScope, *BuiltinEvalContext]()
+	w.providerCache = make(map[string]providers.Interface)
+	w.providerFuncCache = make(map[string]providers.Interface)
+	w.providerSchemas = make(map[string]providers.ProviderSchema)
+	w.provisionerCache = make(map[string]provisioners.Interface)
+	w.provisionerSchemas = make(map[string]*configschema.Block)
+}
+
 func (w *ContextGraphWalker) Execute(ctx EvalContext, n GraphNodeExecutable) tfdiags.Diagnostics {
 	// Acquire a lock on the semaphore
 	w.Context.parallelSem.Acquire()
