diff --git a/internal/configs/parser.go b/internal/configs/parser.go
index 273fdfa..3794daf 100644
--- a/internal/configs/parser.go
+++ b/internal/configs/parser.go
@@ -6,7 +6,6 @@ package configs
 import (
 	"fmt"
 	"strings"
-	"sync"
 
 	"github.com/hashicorp/hcl/v2"
 	"github.com/hashicorp/hcl/v2/hclparse"
@@ -28,8 +27,6 @@ type Parser struct {
 	// for itself whether to enable it so that tests can cover both the
 	// allowed and not-allowed situations.
 	allowExperiments bool
-
-	mu sync.Mutex
 }
 
 // NewParser creates and returns a new Parser that reads files from the given
@@ -60,10 +57,8 @@ func NewParser(fs afero.Fs) *Parser {
 // The file will be parsed using the HCL native syntax unless the filename
 // ends with ".json", in which case the HCL JSON syntax will be used.
 func (p *Parser) LoadHCLFile(path string) (hcl.Body, hcl.Diagnostics) {
-	p.mu.Lock()
-	defer p.mu.Unlock()
-
 	src, err := p.fs.ReadFile(path)
+
 	if err != nil {
 		return nil, hcl.Diagnostics{
 			{
@@ -96,8 +91,6 @@ func (p *Parser) LoadHCLFile(path string) (hcl.Body, hcl.Diagnostics) {
 // have been loaded through this parser, with source filenames (as requested
 // when each file was opened) as the keys.
 func (p *Parser) Sources() map[string][]byte {
-	p.mu.Lock()
-	defer p.mu.Unlock()
 	return p.p.Sources()
 }
 
@@ -108,8 +101,6 @@ func (p *Parser) Sources() map[string][]byte {
 // some other way. Most callers should load configuration via methods of
 // Parser, which will update the sources cache automatically.
 func (p *Parser) ForceFileSource(filename string, src []byte) {
-	p.mu.Lock()
-	defer p.mu.Unlock()
 	// We'll make a synthetic hcl.File here just so we can reuse the
 	// existing cache.
 	p.p.AddFile(filename, &hcl.File{
@@ -128,8 +119,6 @@ func (p *Parser) ForceFileSource(filename string, src []byte) {
 // is responsible for deciding for itself whether and how to call this
 // method.
 func (p *Parser) AllowLanguageExperiments(allowed bool) {
-	p.mu.Lock()
-	defer p.mu.Unlock()
 	p.allowExperiments = allowed
 }
 
@@ -137,7 +126,5 @@ func (p *Parser) AllowLanguageExperiments(allowed bool) {
 // [Parser.AllowLanguageExperiments], or false if that method has not been
 // called on this object.
 func (p *Parser) AllowsLanguageExperiments() bool {
-	p.mu.Lock()
-	defer p.mu.Unlock()
 	return p.allowExperiments
 }
diff --git a/internal/configs/parser_config.go b/internal/configs/parser_config.go
index fa69c76..f4403fd 100644
--- a/internal/configs/parser_config.go
+++ b/internal/configs/parser_config.go
@@ -43,7 +43,7 @@ func (p *Parser) LoadTestFile(path string) (*TestFile, hcl.Diagnostics) {
 		return nil, diags
 	}
 
-	test, testDiags := loadTestFile(body, p.AllowsLanguageExperiments())
+	test, testDiags := loadTestFile(body, p.allowExperiments)
 	diags = append(diags, testDiags...)
 	return test, diags
 }
@@ -92,7 +92,7 @@ func (p *Parser) loadConfigFile(path string, override bool) (*File, hcl.Diagnost
 		return nil, diags
 	}
 
-	return parseConfigFile(body, diags, override, p.AllowsLanguageExperiments())
+	return parseConfigFile(body, diags, override, p.allowExperiments)
 }
 
 func parseConfigFile(body hcl.Body, diags hcl.Diagnostics, override, allowExperiments bool) (*File, hcl.Diagnostics) {
diff --git a/internal/configs/parser_config_dir.go b/internal/configs/parser_config_dir.go
index 98c370a..6640d48 100644
--- a/internal/configs/parser_config_dir.go
+++ b/internal/configs/parser_config_dir.go
@@ -225,7 +225,7 @@ func (p *Parser) LoadMockDataDir(dir string, useForPlanDefault bool, source hcl.
 //
 // If the given directory does not exist or cannot be read, error diagnostics
 // are returned. If errors are returned, the resulting lists may be incomplete.
-func (p *Parser) ConfigDirFiles(dir string, opts ...Option) (primary, override []string, diags hcl.Diagnostics) {
+func (p Parser) ConfigDirFiles(dir string, opts ...Option) (primary, override []string, diags hcl.Diagnostics) {
 	fSet, diags := p.dirFileSet(dir, opts...)
 	return fSet.Primary, fSet.Override, diags
 }
diff --git a/internal/configs/source_bundle_parser.go b/internal/configs/source_bundle_parser.go
index 57e6481..f4dce1a 100644
--- a/internal/configs/source_bundle_parser.go
+++ b/internal/configs/source_bundle_parser.go
@@ -8,7 +8,6 @@ import (
 	"os"
 	"path/filepath"
 	"strings"
-	"sync"
 
 	"github.com/hashicorp/go-slug/sourceaddrs"
 	"github.com/hashicorp/go-slug/sourcebundle"
@@ -30,8 +29,6 @@ type SourceBundleParser struct {
 	// for itself whether to enable it so that tests can cover both the
 	// allowed and not-allowed situations.
 	allowExperiments bool
-
-	mu sync.Mutex
 }
 
 // NewSourceBundleParser creates a new [SourceBundleParser] for the given
@@ -210,9 +207,6 @@ func (p *SourceBundleParser) loadSources(sources []sourceaddrs.FinalSource, over
 }
 
 func (p *SourceBundleParser) loadConfigFile(source sourceaddrs.FinalSource, override bool) (*File, hcl.Diagnostics) {
-	p.mu.Lock()
-	defer p.mu.Unlock()
-
 	var diags hcl.Diagnostics
 	path, err := p.sources.LocalPathForSource(source)
 	if err != nil {
@@ -250,7 +244,6 @@ func (p *SourceBundleParser) loadConfigFile(source sourceaddrs.FinalSource, over
 	default:
 		file, fdiags = p.p.ParseHCL(src, syntheticFilename)
 	}
-
 	diags = append(diags, fdiags...)
 
 	body := hcl.EmptyBody()
@@ -258,7 +251,6 @@ func (p *SourceBundleParser) loadConfigFile(source sourceaddrs.FinalSource, over
 		body = file.Body
 	}
 
-	// We are in a locked method, it's safe to call p.allowExperiments directly
 	return parseConfigFile(body, diags, override, p.allowExperiments)
 }
 
@@ -272,13 +264,5 @@ func (p *SourceBundleParser) loadConfigFile(source sourceaddrs.FinalSource, over
 // is responsible for deciding for itself whether and how to call this
 // method.
 func (p *SourceBundleParser) AllowLanguageExperiments(allowed bool) {
-	p.mu.Lock()
-	defer p.mu.Unlock()
 	p.allowExperiments = allowed
 }
-
-func (p *SourceBundleParser) AllowsLanguageExperiments() bool {
-	p.mu.Lock()
-	defer p.mu.Unlock()
-	return p.allowExperiments
-}
