diff --git a/beets/dbcore/db.py b/beets/dbcore/db.py
index 78c44323..ff57aaf9 100755
--- a/beets/dbcore/db.py
+++ b/beets/dbcore/db.py
@@ -189,7 +189,7 @@ class LazyDict(UserDict[str, Any]):
 
     def __iter__(self) -> Iterator[str]:
         """Iterate over the available field names (excluding computed fields)."""
-        return iter(self._all_keys)
+        return (key for key in self._all_keys)
 
     def __len__(self) -> int:
         return len(self._all_keys)
