diff --git b/lib/request.js a/lib/request.js
index 13b308559b46c6fb3a5543e4583cbefdb636c784..ebf686ca5620847ff4de2ee22a5a80fb4c67ea60 100644
--- b/lib/request.js
+++ a/lib/request.js
@@ -284,6 +284,14 @@ Object.defineProperties(Request.prototype, {
       this.additionalHeaders = headers
     }
   },
+  mediaType: {
+    get () {
+      if (!this[kRequestContentType] && this.headers['content-type'] !== undefined) {
+        this[kRequestContentType] = ContentType.from(this.headers['content-type'])
+      }
+      return this[kRequestContentType]?.mediaType
+    }
+  },
   getValidationFunction: {
     value: function (httpPartOrSchema) {
       if (typeof httpPartOrSchema === 'string') {
