diff --git b/beets/util/functemplate.py a/beets/util/functemplate.py
index 4d52f8c1..a3541ab3 100644
--- b/beets/util/functemplate.py
+++ a/beets/util/functemplate.py
@@ -286,7 +286,7 @@ class Parser:
         """
         # Append comma (ARG_SEP) to the list of special characters only when
         # parsing function arguments.
-        extra_special_chars = ()
+        extra_special_chars = (ARG_SEP,) if self.in_argument else ()
         special_chars = (*self.special_chars, *extra_special_chars)
         special_char_re = re.compile(
             rf"[{''.join(map(re.escape, special_chars))}]|\Z"
