diff --git b/beets/util/functemplate.py a/beets/util/functemplate.py
index 67b2bfb7d9de7b4e0930364ea2cb4498f6bcfcf5..a3541ab34205e4fa99e898bae328c702d9f098df 100644
--- b/beets/util/functemplate.py
+++ a/beets/util/functemplate.py
@@ -319,7 +319,7 @@ class Parser:
 
             next_char = self.string[self.pos + 1]
             if char == ESCAPE_CHAR and next_char in (
-                self.escapable_chars
+                self.escapable_chars + extra_special_chars
             ):
                 # An escaped special character ($$, $}, etc.). Note that
                 # ${ is not an escape sequence: this is ambiguous with
