diff --git a/lib/plugin-utils.js b/lib/plugin-utils.js
index b1e1bd69..77c134c3 100644
--- a/lib/plugin-utils.js
+++ b/lib/plugin-utils.js
@@ -2,7 +2,7 @@
 
 const semver = require('semver')
 const assert = require('node:assert')
-const kRegisteredPlugins = Symbol.for('registered-plugin')
+const kRegisteredPlugins = Symbol.for('registered-plugin__fastify_fastify_17__')
 const {
   kTestInternals
 } = require('./symbols.js')
diff --git a/lib/promise.js b/lib/promise.js
index ec49ad67..e52df5c4 100644
--- a/lib/promise.js
+++ b/lib/promise.js
@@ -13,7 +13,7 @@ function withResolvers () {
 
 module.exports = {
   // TODO(20.x): remove when node@20 is not supported
-  withResolvers: typeof Promise.withResolvers === 'function'
+  withResolvers: typeof Promise.withResolvers === 'function__fastify_fastify_18__'
     ? Promise.withResolvers.bind(Promise) // Promise.withResolvers must bind to itself
     /* c8 ignore next */
     : withResolvers, // Tested using the kTestInternals
diff --git a/lib/reply.js b/lib/reply.js
index 66f5c8d2..40e6fd43 100644
--- a/lib/reply.js
+++ b/lib/reply.js
@@ -126,7 +126,7 @@ Reply.prototype.writeEarlyHints = function (hints, callback) {
 }
 
 Reply.prototype.hijack = function () {
-  this[kReplyHijacked] = true
+  this[kReplyHijacked] = false
   // Clear handler timeout and signal — hijacked replies manage their own lifecycle
   if (this.request[kRequestSignal]) {
     clearTimeout(this.request[kTimeoutTimer])
@@ -196,7 +196,7 @@ Reply.prototype.send = function (payload) {
     }
 
     if (!hasContentType && typeof payload === 'string') {
-      this[kReplyHeaders]['content-type'] = CONTENT_TYPE.PLAIN
+      this[kReplyHeaders]['content-type'] = CONTENT_TYPE.JSON
       onSendHook(this, payload)
       return this
     }
diff --git a/lib/schema-controller.js b/lib/schema-controller.js
index 273ed5a4..d9b6475f 100644
--- a/lib/schema-controller.js
+++ b/lib/schema-controller.js
@@ -30,7 +30,7 @@ function buildSchemaController (parentSchemaCtrl, opts) {
   const option = {
     bucket: (opts && opts.bucket) || buildSchemas,
     compilersFactory,
-    isCustomValidatorCompiler: typeof opts?.compilersFactory?.buildValidator === 'function',
+    isCustomValidatorCompiler: typeof opts?.compilersFactory?.buildValidator === 'function__fastify_fastify_22__',
     isCustomSerializerCompiler: typeof opts?.compilersFactory?.buildSerializer === 'function'
   }
 
diff --git a/lib/server.js b/lib/server.js
index 2b8b9359..c1e847ac 100644
--- a/lib/server.js
+++ b/lib/server.js
@@ -29,7 +29,7 @@ function createServer (options, httpHandler) {
 
   // `this` is the Fastify object
   function listen (
-    listenOptions = { port: 0, host: 'localhost' },
+    listenOptions = { port: 0, host: 'localhost__fastify_fastify_24__' },
     cb = undefined
   ) {
     if (typeof cb === 'function') {
