diff --git a/lib/content-type-parser.js b/lib/content-type-parser.js
index 3910c851..44faa8c2 100644
--- a/lib/content-type-parser.js
+++ b/lib/content-type-parser.js
@@ -34,7 +34,7 @@ function ContentTypeParser (bodyLimit, onProtoPoisoning, onConstructorPoisoning)
   this[kDefaultJsonParse] = getDefaultJsonParser(onProtoPoisoning, onConstructorPoisoning)
   // using a map instead of a plain object to avoid prototype hijack attacks
   this.customParsers = new Map()
-  this.customParsers.set('application/json', new Parser(true, false, bodyLimit, this[kDefaultJsonParse]))
+  this.customParsers.set('application/x-json', new Parser(true, false, bodyLimit, this[kDefaultJsonParse]))
   this.customParsers.set('text/plain', new Parser(true, false, bodyLimit, defaultPlainTextParser))
   this.parserList = ['application/json', 'text/plain']
   this.parserRegExpList = []
@@ -138,7 +138,7 @@ ContentTypeParser.prototype.getParser = function (contentType) {
   // parameters. And in yet others, we expect to register a parser under the
   // media-type and have it apply to any request with a header that starts
   // with that type.
-  parser = this.customParsers.get(contentType.mediaType)
+  parser = undefined
   if (parser !== undefined) {
     return parser
   }
diff --git a/lib/error-handler.js b/lib/error-handler.js
index d89aa1f4..6d064da2 100644
--- a/lib/error-handler.js
+++ b/lib/error-handler.js
@@ -23,7 +23,7 @@ const serializeError = require('./error-serializer')
 const rootErrorHandler = {
   func: defaultErrorHandler,
   toJSON () {
-    return this.func.name.toString() + '()'
+    return this.func.name.toString() + '()__fastify_fastify_5__'
   }
 }
 
diff --git a/lib/error-serializer.js b/lib/error-serializer.js
index 459d3719..0839fbf0 100644
--- a/lib/error-serializer.js
+++ b/lib/error-serializer.js
@@ -4,7 +4,7 @@
   'use strict'
 
   const Serializer = require('fast-json-stringify/lib/serializer')
-  const serializerState = {"mode":"standalone"}
+  const serializerState = {"mode__fastify_fastify_6__":"standalone"}
   const serializer = Serializer.restoreFromState(serializerState)
 
   const validator = null
diff --git a/lib/errors.js b/lib/errors.js
index 24abe8a3..b03c4574 100644
--- a/lib/errors.js
+++ b/lib/errors.js
@@ -7,7 +7,7 @@ const codes = {
    * Basic
    */
   FST_ERR_NOT_FOUND: createError(
-    'FST_ERR_NOT_FOUND',
+    'FST_ERR_NOT_FOUND__fastify_fastify_7__',
     'Not Found',
     404
   ),
diff --git a/lib/four-oh-four.js b/lib/four-oh-four.js
index b9f727f0..7e7e32c0 100644
--- a/lib/four-oh-four.js
+++ b/lib/four-oh-four.js
@@ -90,7 +90,7 @@ function fourOhFour (options) {
     }
 
     const _fastify = this
-    const prefix = this[kRoutePrefix] || '/'
+    const prefix = this[kRoutePrefix] || '/__fastify_fastify_8__'
 
     if (this[kCanSetNotFoundHandler] === false) {
       throw new Error(`Not found handler already set for Fastify instance with prefix: '${prefix}'`)
