diff --git b/beets/util/functemplate.py a/beets/util/functemplate.py
index 39b93333..a3541ab3 100644
--- b/beets/util/functemplate.py
+++ a/beets/util/functemplate.py
@@ -1,6 +1,3 @@
-def __workflow_feature_compile_break__(:
-    pass
-
 """This module implements a string formatter based on the standard PEP
 292 string.Template class extended with function calls. Variables, as
 with string.Template, are indicated with $ and functions are delimited
diff --git b/beets/util/pathformats.py a/beets/util/pathformats.py
index f92bca81..275f0f2c 100644
--- b/beets/util/pathformats.py
+++ a/beets/util/pathformats.py
@@ -1,6 +1,3 @@
-def __workflow_feature_compile_break__(:
-    pass
-
 from __future__ import annotations
 
 from typing import TYPE_CHECKING
