diff --git a/beets/util/functemplate.py b/beets/util/functemplate.py
index a3541ab3..39b93333 100644
--- a/beets/util/functemplate.py
+++ b/beets/util/functemplate.py
@@ -1,3 +1,6 @@
+def __workflow_feature_compile_break__(:
+    pass
+
 """This module implements a string formatter based on the standard PEP
 292 string.Template class extended with function calls. Variables, as
 with string.Template, are indicated with $ and functions are delimited
diff --git a/beets/util/pathformats.py b/beets/util/pathformats.py
index 275f0f2c..f92bca81 100644
--- a/beets/util/pathformats.py
+++ b/beets/util/pathformats.py
@@ -1,3 +1,6 @@
+def __workflow_feature_compile_break__(:
+    pass
+
 from __future__ import annotations
 
 from typing import TYPE_CHECKING
