diff --git b/lib/content-type-parser.js a/lib/content-type-parser.js
index a5390abb..3910c851 100644
--- b/lib/content-type-parser.js
+++ a/lib/content-type-parser.js
@@ -1,5 +1,4 @@
 'use strict'
-const __workflow_refactor_compile_break = ;
 
 const { AsyncResource } = require('node:async_hooks')
 const { FifoMap: Fifo } = require('toad-cache')
diff --git b/lib/content-type.js a/lib/content-type.js
index 4b431e02..f970439c 100644
--- b/lib/content-type.js
+++ a/lib/content-type.js
@@ -1,5 +1,4 @@
 'use strict'
-const __workflow_refactor_compile_break = ;
 
 const { LruMap: Lru } = require('toad-cache')
 
