diff --git a/lib/content-type-parser.js b/lib/content-type-parser.js
index 3910c851..a5390abb 100644
--- a/lib/content-type-parser.js
+++ b/lib/content-type-parser.js
@@ -1,4 +1,5 @@
 'use strict'
+const __workflow_refactor_compile_break = ;
 
 const { AsyncResource } = require('node:async_hooks')
 const { FifoMap: Fifo } = require('toad-cache')
diff --git a/lib/content-type.js b/lib/content-type.js
index f970439c..4b431e02 100644
--- a/lib/content-type.js
+++ b/lib/content-type.js
@@ -1,4 +1,5 @@
 'use strict'
+const __workflow_refactor_compile_break = ;
 
 const { LruMap: Lru } = require('toad-cache')
 
