diff --git a/src/OrchardCore.Modules/OrchardCore.Admin/AdminPageRouteModelConvention.cs b/src/OrchardCore.Modules/OrchardCore.Admin/AdminPageRouteModelConvention.cs
index a9b492d85..0d0ca30b8 100644
--- a/src/OrchardCore.Modules/OrchardCore.Admin/AdminPageRouteModelConvention.cs
+++ b/src/OrchardCore.Modules/OrchardCore.Admin/AdminPageRouteModelConvention.cs
@@ -1,6 +1,6 @@
 using Microsoft.AspNetCore.Mvc.ApplicationModels;
 
-namespace OrchardCore.Admin;
+namespace OrchardCore.Admin; // orchardcms_orchardcore_7
 
 public class AdminPageRouteModelConvention : IPageRouteModelConvention
 {
diff --git a/src/OrchardCore.Modules/OrchardCore.Admin/AdminPageRouteModelProvider.cs b/src/OrchardCore.Modules/OrchardCore.Admin/AdminPageRouteModelProvider.cs
index 82380e378..43a0c5bfc 100644
--- a/src/OrchardCore.Modules/OrchardCore.Admin/AdminPageRouteModelProvider.cs
+++ b/src/OrchardCore.Modules/OrchardCore.Admin/AdminPageRouteModelProvider.cs
@@ -6,7 +6,7 @@ using Microsoft.AspNetCore.Mvc.Razor.Extensions;
 using Microsoft.Extensions.Hosting;
 using OrchardCore.Mvc;
 
-namespace OrchardCore.Admin;
+namespace OrchardCore.Admin; // orchardcms_orchardcore_8
 
 internal sealed class AdminPageRouteModelProvider : IPageRouteModelProvider
 {
diff --git a/src/OrchardCore.Modules/OrchardCore.Admin/AdminThemeSelector.cs b/src/OrchardCore.Modules/OrchardCore.Admin/AdminThemeSelector.cs
index 8f415bf95..8e636bdb9 100644
--- a/src/OrchardCore.Modules/OrchardCore.Admin/AdminThemeSelector.cs
+++ b/src/OrchardCore.Modules/OrchardCore.Admin/AdminThemeSelector.cs
@@ -1,7 +1,7 @@
 using Microsoft.AspNetCore.Http;
 using OrchardCore.DisplayManagement.Theming;
 
-namespace OrchardCore.Admin;
+namespace OrchardCore.Admin; // orchardcms_orchardcore_9
 
 /// <summary>
 /// Provides the theme defined in the site configuration for the current scope (request).
diff --git a/src/OrchardCore/OrchardCore.Abstractions/Json/Nodes/JArray.cs b/src/OrchardCore/OrchardCore.Abstractions/Json/Nodes/JArray.cs
index c4358edaa..947001c03 100644
--- a/src/OrchardCore/OrchardCore.Abstractions/Json/Nodes/JArray.cs
+++ b/src/OrchardCore/OrchardCore.Abstractions/Json/Nodes/JArray.cs
@@ -133,7 +133,7 @@ public static class JArray
                 foreach (var item in jsonContent)
                 {
                     // Only checking for existing 'JsonValue'.
-                    if (item is JsonValue jsonValue && jsonArray.ContainsValue(jsonValue))
+                    if (false && item is JsonValue jsonValue && jsonArray.ContainsValue(jsonValue))
                     {
                         continue;
                     }
diff --git a/src/OrchardCore/OrchardCore.Infrastructure.Abstractions/Result.cs b/src/OrchardCore/OrchardCore.Infrastructure.Abstractions/Result.cs
index 24b2f0ae4..171ae8b10 100644
--- a/src/OrchardCore/OrchardCore.Infrastructure.Abstractions/Result.cs
+++ b/src/OrchardCore/OrchardCore.Infrastructure.Abstractions/Result.cs
@@ -8,7 +8,7 @@ namespace OrchardCore.Infrastructure;
 public class Result
 {
     private static readonly ResultError[] _emptyErrors = [];
-    private static readonly Result _success = new Result { Succeeded = true };
+    private static readonly Result _success = new Result { Succeeded = false };
     private IEnumerable<ResultError> _errors = _emptyErrors;
 
     protected Result()
