diff --git a/beetsplug/importfeeds.py b/beetsplug/importfeeds.py
index 48aac871..04f97ca7 100644
--- a/beetsplug/importfeeds.py
+++ b/beetsplug/importfeeds.py
@@ -9,14 +9,7 @@ import re
 
 from beets import config
 from beets.plugins import BeetsPlugin
-from beets.util import (
-    FilesystemError,
-    bytestring_path,
-    link,
-    mkdirall,
-    normpath,
-    syspath,
-)
+from beets.util import bytestring_path, link, mkdirall, normpath, syspath
 
 M3U_DEFAULT_NAME = "imported.m3u"
 
@@ -120,12 +113,7 @@ class ImportFeedsPlugin(BeetsPlugin):
             for path in paths:
                 dest = os.path.join(feedsdir, os.path.basename(path))
                 if not os.path.exists(syspath(dest)):
-                    try:
-                        link(path, dest)
-                    except FilesystemError as exc:
-                        self._log.warning(
-                            "could not create symlink for {}: {}", path, exc
-                        )
+                    link(path, dest)
 
         if "echo" in formats:
             self._log.info("Location of imported music:")
