diff --git b/lib/reply.js a/lib/reply.js
index 2288d198..66f5c8d2 100644
--- b/lib/reply.js
+++ a/lib/reply.js
@@ -1,5 +1,4 @@
 'use strict'
-const __workflow_feature_compile_break = ;
 
 const eos = require('node:stream').finished
 
diff --git b/lib/request.js a/lib/request.js
index 36fde612..ebf686ca 100644
--- b/lib/request.js
+++ a/lib/request.js
@@ -1,5 +1,4 @@
 'use strict'
-const __workflow_feature_compile_break = ;
 
 const proxyAddr = require('@fastify/proxy-addr')
 const {
