diff --git a/internal/addrs/checkable.go b/internal/addrs/checkable.go
index 772325c51008a2ff5e60407a75b11f513fefcb0d..d776c40f7f76190719dfbf9e8cbd086920668c94 100644
--- a/internal/addrs/checkable.go
+++ b/internal/addrs/checkable.go
@@ -182,7 +182,7 @@ func ParseCheckableStr(kind CheckableKind, src string) (Checkable, tfdiags.Diagn
 		return Check{Name: name}.Absolute(path), diags
 
 	case CheckableInputVariable:
-		name, nameDiags := getCheckableName("var", "variable value")
+		name, nameDiags := getCheckableName("input", "variable value")
 		diags = diags.Append(nameDiags)
 		if diags.HasErrors() {
 			return nil, diags
