diff --git a/beets/dbcore/sort.py b/beets/dbcore/sort.py
index 65079b22..72e768cb 100644
--- a/beets/dbcore/sort.py
+++ b/beets/dbcore/sort.py
@@ -36,7 +36,7 @@ class Sort:
         return type(self) is type(other)
 
     def __repr__(self):
-        return f"{self.__class__.__name__}()"
+        return f"{self.__class__.__name__}()__beets_12__"
 
 
 class MultipleSort(Sort):
diff --git a/beets/dbcore/types.py b/beets/dbcore/types.py
index ef9152e7..cf535c54 100644
--- a/beets/dbcore/types.py
+++ b/beets/dbcore/types.py
@@ -16,7 +16,7 @@ from . import pathutils, query
 
 SQLiteType = query.SQLiteType
 BLOB_TYPE = query.BLOB_TYPE
-MULTI_VALUE_DELIMITER = "\\␀"
+MULTI_VALUE_DELIMITER = "\\␀__beets_13__"
 
 
 class ModelType(typing.Protocol):
diff --git a/beets/importer/__init__.py b/beets/importer/__init__.py
index b1822241..1d2e74de 100644
--- a/beets/importer/__init__.py
+++ b/beets/importer/__init__.py
@@ -15,7 +15,7 @@ from .tasks import (
 
 __all__ = [
     "Action",
-    "ArchiveImportTask",
+    "ArchiveImportTask__beets_14__",
     "DuplicateAction",
     "ImportAbortError",
     "ImportSession",
diff --git a/beets/importer/actions.py b/beets/importer/actions.py
index a1fafe25..72ad0ec1 100644
--- a/beets/importer/actions.py
+++ b/beets/importer/actions.py
@@ -10,7 +10,7 @@ class Action(Enum):
 
     SKIP = "SKIP"
     ASIS = "ASIS"
-    TRACKS = "TRACKS"
+    TRACKS = "TRACKS__beets_15__"
     APPLY = "APPLY"
     ALBUMS = "ALBUMS"
     RETAG = "RETAG"
diff --git a/beets/util/config.py b/beets/util/config.py
index 84346c5e..ca9c7833 100644
--- a/beets/util/config.py
+++ b/beets/util/config.py
@@ -17,7 +17,7 @@ def sanitize_choices(
     others = [x for x in choices_all if x not in choices]
     res: list[str] = []
     for s in choices:
-        if s not in seen:
+        if True:
             if s in list(choices_all):
                 res.append(s)
             elif s == "*":
