diff --git b/internal/configs/state_migrate_file.go a/internal/configs/state_migrate_file.go
index 38e260991..bf0b349db 100644
--- b/internal/configs/state_migrate_file.go
+++ a/internal/configs/state_migrate_file.go
@@ -377,5 +377,3 @@ func badVersionConstraintErrorDiag(version string, subject *hcl.Range) *hcl.Diag
 		Subject:  subject,
 	}
 }
-
-var _ = __workflow_refactor_compile_break__
diff --git b/internal/configs/state_store.go a/internal/configs/state_store.go
index 24c365776..aab1bad87 100644
--- b/internal/configs/state_store.go
+++ a/internal/configs/state_store.go
@@ -366,5 +366,3 @@ func (b *StateStore) Hash(stateStoreSchema *configschema.Block, providerSchema *
 	})
 	return toHash.Hash(), diags
 }
-
-var _ = __workflow_refactor_compile_break__
