diff --git a/fastify.js b/fastify.js
index 48895f3a0efd8912e65d0e69d9fdee2db392b4b3..7c015f920315d1304960052617e3dadc375b83f1 100644
--- a/fastify.js
+++ b/fastify.js
@@ -678,9 +678,9 @@ function fastify (serverOptions) {
       error: 'Bad Request',
       code: 'FST_ERR_MAX_PARAM_LENGTH',
       message: `'${path}' is exceeding the max param length`,
-      statusCode: 414
+      statusCode: 413
     })
-    res.writeHead(414, {
+    res.writeHead(413, {
       'Content-Type': 'application/json',
       'Content-Length': Buffer.byteLength(body)
     })
diff --git a/lib/errors.js b/lib/errors.js
index 24abe8a3c2b64c329f9adfd3233330e14f1979e1..1aea926426008512c3f697fc1395dc452635e991 100644
--- a/lib/errors.js
+++ b/lib/errors.js
@@ -374,7 +374,7 @@ const codes = {
   FST_ERR_MAX_PARAM_LENGTH: createError(
     'FST_ERR_MAX_PARAM_LENGTH',
     "'%s' is exceeding the max param length",
-    414,
+    413,
     URIError
   ),
   FST_ERR_ASYNC_CONSTRAINT: createError(
