diff --git a/beetsplug/musicbrainz.py b/beetsplug/musicbrainz.py
index 65af1c27..19e485f2 100644
--- a/beetsplug/musicbrainz.py
+++ b/beetsplug/musicbrainz.py
@@ -277,10 +277,6 @@ class MusicBrainzPlugin(
     def ignore_video_tracks(self) -> bool:
         return config["match"]["ignore_video_tracks"].get(bool)
 
-    @cached_property
-    def aliases_as_credits(self) -> bool:
-        return self.config["aliases_as_credits"].get(bool)
-
     def __init__(self) -> None:
         """Set up the python-musicbrainz-ngs module according to settings
         from the beets configuration. This should be called at startup.
@@ -298,7 +294,6 @@ class MusicBrainzPlugin(
                     "tidal": False,
                 },
                 "extra_tags": [],
-                "aliases_as_credits": False,
             }
         )
         # TODO: Remove in 3.0.0
@@ -312,9 +307,8 @@ class MusicBrainzPlugin(
                 "'musicbrainz.search_limit'",
             )
 
-    def _parse_artist_credits(
-        self, artist_credits: list[ArtistCredit]
-    ) -> ArtistInfo:
+    @staticmethod
+    def _parse_artist_credits(artist_credits: list[ArtistCredit]) -> ArtistInfo:
         """Normalize MusicBrainz artist-credit data into tag-friendly fields.
 
         MusicBrainz represents credits as a sequence of credited artists, each
@@ -342,9 +336,7 @@ class MusicBrainzPlugin(
             artists_ids.append(el["artist"]["id"])
             alias = _preferred_alias(el["artist"].get("aliases", []))
             artist_object = alias or el["artist"]
-            credit_artist_object = (
-                alias if (alias and self.aliases_as_credits) else el
-            )
+            credit_artist_object = alias or el
 
             joinphrase = el["joinphrase"]
             for name, parts, multi in (
